core/src/main/java/org/apache/iceberg/util/ThreadPools.java

```diff
@@ -38,6 +38,9 @@ private ThreadPools() {
       WORKER_THREAD_POOL_SIZE_PROP,
       Runtime.getRuntime().availableProcessors());
 
+  // In default, we submit 2 tasks per worker at a time.
+  public static final int WORKER_THREAD_POOL_PARALLELISM = WORKER_THREAD_POOL_SIZE * 2;
```

Comment:
This doesn't make sense here and complicates the `ParallelIterator` API. I think the parallel iterator should just double the parallelism and document that it expects the parallelism to be the pool size.